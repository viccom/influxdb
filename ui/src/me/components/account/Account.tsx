// Libraries
import React, {PureComponent} from 'react'
import {WithRouterProps} from 'react-router'

// Components
import {Page} from 'src/pageLayout'
import TabbedPage from 'src/shared/components/tabbed_page/TabbedPage'
import TabbedPageSection from 'src/shared/components/tabbed_page/TabbedPageSection'
import Settings from 'src/me/components/account/Settings'
import Tokens from 'src/me/components/account/Tokens'

export enum Tabs {
  Settings = 'settings',
  Tokens = 'tokens',
}

export default class Account extends PureComponent<WithRouterProps> {
  public render() {
    const {params} = this.props
    return (
      <Page titleTag="Account">
        <Page.Header fullWidth={false}>
          <Page.Header.Left>
            <Page.Title title="Account" />
          </Page.Header.Left>
          <Page.Header.Right />
        </Page.Header>
        <Page.Contents fullWidth={false} scrollable={true}>
          <TabbedPage
            name="Account"
            activeTabUrl={params.tab}
            parentUrl="/account"
          >
            <TabbedPageSection
              title="My Settings"
              id={Tabs.Settings}
              url={Tabs.Settings}
            >
              <Settings />
            </TabbedPageSection>
            <TabbedPageSection
              title="My Tokens"
              id={Tabs.Tokens}
              url={Tabs.Tokens}
            >
              <Tokens />
            </TabbedPageSection>
          </TabbedPage>
        </Page.Contents>
      </Page>
    )
  }
}
