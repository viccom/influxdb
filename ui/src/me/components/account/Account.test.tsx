// Libraries
import React from 'react'
import {shallow} from 'enzyme'

// Components
import Account from 'src/me/components/account/Account'

const setup = (override?) => {
  const props = {
    params: {tab: 'settings'},
    ...override,
  }

  const wrapper = shallow(<Account {...props} />)

  return {wrapper}
}

describe('Account', () => {
  describe('rendering', () => {
    it('renders!', () => {
      const {wrapper} = setup()

      expect(wrapper.exists()).toBe(true)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
