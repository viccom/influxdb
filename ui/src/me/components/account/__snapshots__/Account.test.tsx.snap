// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account rendering renders! 1`] = `
<Page
  titleTag="Account"
>
  <PageHeader
    fullWidth={false}
    inPresentationMode={false}
  >
    <PageHeaderLeft
      offsetPixels={0}
    >
      <PageTitle
        title="Account"
      />
    </PageHeaderLeft>
    <PageHeaderRight
      offsetPixels={0}
    />
  </PageHeader>
  <PageContents
    fullWidth={false}
    scrollable={true}
  >
    <withRouter(TabbedPage)
      activeTabUrl="settings"
      name="Account"
      parentUrl="/account"
    >
      <TabbedPageSection
        id="settings"
        title="My Settings"
        url="settings"
      >
        <Connect(Settings) />
      </TabbedPageSection>
      <TabbedPageSection
        id="tokens"
        title="My Tokens"
        url="tokens"
      >
        <Connect(Tokens) />
      </TabbedPageSection>
    </withRouter(TabbedPage)>
  </PageContents>
</Page>
`;
